### PR TITLE
Add compress_pickle support to PickleDataSet

### PR DESCRIPTION
**Pull request by takeru**
## Description
Add compress_pickle support to PickleDataSet.

## Development notes

Add import/BACKENDS like a "joblib".
Tested to save/load .lz4 in ipython.

## Checklist

- [x] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [x] Added tests to cover my changes

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -18,11 +18,12 @@
 * Fixed `kedro install` for an Anaconda environment defined in `environment.yml`.
 * Fixed backwards compatibility with templates generated with older Kedro versions <0.16.5. No longer need to update `.kedro.yml` to use `kedro lint` and `kedro jupyter notebook convert`.
 * Improved documentation.
+* Added compress_pickle support to PickleDataSet.
```
Comment (lorenabalan) [inline RELEASE.md:21]:
```suggestion
* Added support for `compress_pickle` backend to `PickleDataSet`.
```

File: kedro/extras/datasets/pickle/pickle_dataset.py [hunk 2]
```diff
@@ -51,6 +51,11 @@
 except ImportError:  # pragma: no cover
     joblib = None
 
+try:
+    import compress_pickle
```
Comment (lorenabalan) [inline kedro/extras/datasets/pickle/pickle_dataset.py:55]: Can you also update the docstrings on line 31 please? :)

**Review by lorenabalan — COMMENTED** [review]
Interesting find. Could you please also add the package to our `test_requirements.txt`?
It also feels like we should parameterise the tests so we cover behaviour for all three backends, not just `pickle`. WDYT?

File: kedro/extras/datasets/pickle/pickle_dataset.py (hunk 2 above)
Comment (takeru) [reply kedro/extras/datasets/pickle/pickle_dataset.py:55]: Done! Thank you.

File: kedro/extras/datasets/pickle/pickle_dataset.py [hunk 3]
```diff
@@ -51,6 +51,11 @@
...
+except ImportError:  # pragma: no cover
```
Comment (andrii-ivaniuk) [inline kedro/extras/datasets/pickle/pickle_dataset.py:56]: Should we use `ModuleNotFoundError` here?

@lorenabalan what do you think?

File: tests/extras/datasets/pickle/test_pickle_dataset.py [hunk 4]
```diff
@@ -148,7 +163,7 @@ def test_unserializable_data(self, pickle_data_set, dummy_dataframe, mocker):
             pickle_data_set.save(dummy_dataframe)
 
     def test_invalid_backend(self):
-        pattern = r"'backend' should be one of \['pickle', 'joblib'\], got 'invalid'\."
+        pattern = r"'backend' should be one of \['pickle', 'joblib', 'compress_pickle'\], got 'invalid'\."  # pylint: disable=line-too-long
```
Comment (andrii-ivaniuk) [inline tests/extras/datasets/pickle/test_pickle_dataset.py:166]:
```suggestion
        pattern = (
            r"'backend' should be one of \['pickle', 'joblib', 'compress_pickle'\], "
            r"got 'invalid'\."
        )
```

**Comment by andrii-ivaniuk** [comment]
How about adding [dill](https://pypi.org/project/dill/) as a backend lib as well?
@lorenabalan WDYT?

**Comment by lorenabalan** [comment]
Hi @takeru sorry about the delay. Had a quick chat with the team and it would make more sense if this was a separate dataset class instead of another backend, and specify its dependency in extra requires, similarly to what we do with pandas. Maybe `pickle.CompressPickleDataSet`? We realise there's already an unfortunate precedent with `joblib` in there, so we're planning on refactoring that (in a breaking-change release) using the same mental model.

**Comment by andrii-ivaniuk** [comment]
It was decided to extend the existing dataset by supporting a new backend.

**Comment by andrii-ivaniuk** [comment]
I switched from `master` to `develop` to pick up fixes related to CI.

**Comment by andrii-ivaniuk** [comment]
Switched back to the `master` branch.

**Comment by takeru** [comment]
どうもありがとうございました😊

**Pull request merged**